src/app/todo/services/todo.services.ts

```diff
@@ -11,20 +12,21 @@ export class TaskService {
     this.tasks
   );
 
-  constructor() {}
+  constructor(private taskApi: TaskApiService) {}
 
   getTasks(): Observable<ITask[]> {
-    console.log(this.tasks);
-    return this.tasksSubject.asObservable();
+    return this.taskApi.get('http://localhost:3001/api/todos');
   }
 
   addTask(task: ITask): void {
     this.tasks.push(task);
     this.tasksSubject.next(this.tasks);
   }
-  deleteTask(id: number): void {
-    const updatedTasks = this.tasks.filter((task) => task.id !== id);
-    this.tasks = updatedTasks;
-    this.tasksSubject.next(this.tasks);
+  deleteTask(id: number): Observable<ITask[]> {
+    return this.taskApi.deleteTask('http://localhost:3001/api/todoDelete', id);
+  }
+  getTask(id: number): Observable<ITask | null> {
```

Comment:
I feel like getTask here is useless since you already get the task you need from getTask method in TaskServiceApi